core/src/main/java/io/grpc/internal/DnsNameResolver.java

```diff
@@ -138,17 +138,17 @@
   private final String host;
   private final int port;
   private final Resource<Executor> executorResource;
-  @GuardedBy("this")
+  private final long cacheTtlNanos;
+  private final Stopwatch stopwatch;
+  private final SynchronizationContext syncContext;
+
+  // Following fields must be accessed from synContext
+  private ResolutionResults cachedResolutionResults = null;
   private boolean shutdown;
-  @GuardedBy("this")
   private Executor executor;
-  @GuardedBy("this")
   private boolean resolving;
-  @GuardedBy("this")
   private Listener listener;
```

Comment:
It looks like this can be accessed outside of the syncContext; it's only changed in start().

Reply:
I'm saying that listener can actually be accessed from any thread. It doesn't change.